plugins/module_utils/redfish_utils.py

```diff
@@ -2187,8 +2201,12 @@ def _find_empty_virt_media_slot(resources, media_types,
             else:
                 if media_match_strict:
                     continue
-            # if ejected, 'Inserted' should be False
-            if (not data.get('Inserted', False)):
+            # Base on current Lenovo server capability, filter out slot RDOC1/2 and Remote1/2/3/4 which are not supported to Insert/Eject.
+            if vendor == 'Lenovo' and ('RDOC' in uri or 'Remote' in uri):
```

Comment:
I'm still working to see if there are alternatives to this type of inspection. I'm not coming up on much at the moment though, so this may be the only path forward.

Reply:
Unfortunately I'm not finding other good alternatives, so this is the best we can work with at this time.